java/configuration/src/main/java/sleeper/configuration/properties/SleeperProperties.java

```diff
@@ -62,30 +62,85 @@ protected SleeperProperties(Properties properties) {
         this.properties = properties;
     }
 
-    protected void init() {
+    /**
+     * A method to be called any time the properties are reloaded as a whole. This should perform any operations that
+     * would be done in the constructor, and also revalidate the values. This should be overridden if any metadata needs
+     * to be set at load time.
+     *
+     * @throws SleeperPropertiesInvalidException if any value is invalid
+     */
+    protected void init() throws SleeperPropertiesInvalidException {
         validate();
     }
 
-    public final void validate() {
+    /**
+     * Validates the values all properties.
```

Comment:
Suggestion : "Validates the values of all the properties"

Reply:
Done, thanks